File: agu_chat/lib/models/last_message_chat.dart
class LastMessageChat {
  String id;
  String senderId;
  String senderUsername;
  String recipientId;
  String recipientUsername;
  String content;
  DateTime messageLastDate;
  String groupName;
  bool isRead;
  int unreadCount = 0;
  String senderDisplayName;
  String? senderImgUrl;// fix error type NULL not sub type of String, because senderImgUrl can return null

  LastMessageChat(
      {required this.id,
      required this.senderId,
      required this.senderUsername,
      required this.recipientId,
      required this.recipientUsername,
      required this.content,
      required this.messageLastDate,
      required this.groupName,
      required this.isRead,
        required this.unreadCount,
        required this.senderDisplayName,
        required this.senderImgUrl
      });

  factory LastMessageChat.fromJson(Map<String, dynamic> json) {
    return LastMessageChat(
        id: json['id'],
      senderId: json['senderId'],
      senderUsername: json['senderUsername'],
      recipientId: json['recipientId'],
      recipientUsername: json['recipientUsername'],
      content: json['content'],
      messageLastDate: DateTime.parse(json['messageLastDate']),
      groupName: json['groupName'],
      isRead: json['isRead'],
      unreadCount: 0,
        senderDisplayName: json['senderDisplayName'],
      senderImgUrl: json['senderImgUrl'],
    );
  }
}

File: ReactMXHApi6/ReactMXHApi6/Core/Entities/LastMessageChat.cs
using System.ComponentModel.DataAnnotations;

namespace ReactMXHApi6.Core.Entities
{
    public class LastMessageChat
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();
        public string SenderId { get; set; }
        public string SenderUsername { get; set; }
        public AppUser Sender { get; set; }
        public string RecipientId { get; set; }
        public string RecipientUsername { get; set; }
        public AppUser Recipient { get; set; }
        public string Content { get; set; }
        public DateTime MessageLastDate { get; set; }
        public string GroupName { get; set; }
        public bool IsRead { get; set; } = false;
    }
}
